src/k8s-engine/k3sHelper.ts

```diff
@@ -461,6 +472,129 @@ export default class K3sHelper extends events.EventEmitter {
     checksum: { current: 0, max: 0 },
   };
 
+  /**
+   * Find the cached version closest to the desired version.
+   * @param desiredVersion The semver of the version of k3s the system would prefer to use, with a '+k3s###' suffix
+   * @returns A semver of the version to use, also with a '+k3s###' suffix
+   */
+  async selectClosestImage(desiredVersion: semver.SemVer): Promise<semver.SemVer> {
+    const cacheDir = path.join(paths.cache, 'k3s');
+    const k3sFilenames = (await fs.promises.readdir(cacheDir))
+      .filter(dirname => /^v\d+\.\d+\.\d+\+k3s\d+$/.test(dirname));
+
+    return this.selectClosestSemVer(desiredVersion, k3sFilenames);
+  }
+
+  protected k3sValue(v: semver.SemVer): number {
+    try {
+      return parseInt((v.build[0] as string).replace('k3s', ''), 10) || 0;
+    } catch {
+      return 0;
+    }
+  }
+
+  // A comparator when the versions are the same so we need to compare the numeric part of the '+k3s...' parts
+  protected compareBuildVersions(v1: semver.SemVer, v2: semver.SemVer): number {
+    return this.k3sValue(v1) - this.k3sValue(v2);
+  }
+
+  /**
+   * semver knows how to do correct comparisons with a '-k3s###' suffix[1], but not '+k3s###', so we need to convert.
+   * This function returns a semver that wraps a version with a '+k3s###' suffix.
```

Comment:
This description doesn't sound like it's relevant to the function name at all. One of the two probably needs to be adjusted.  It's also unclear where the `-k3s??` version strings come from; the prefs only keep up to the minor (`v1.2.3`), and the cache files have plus on my system.

Reply:
Again the comments weren't updated. Initially I converted the `+k3s` part of each filename to `-k3s` because I thought `semver` would handle that more natively. But that turned out to offer no advantage, so I backed out that code, but not the comment.